mate_menu/plugins/applications.py

```diff
@@ -1534,15 +1557,18 @@ def updateBoxes( self, menu_has_changed ):
                     else:
                         launcherNames.append(launcherName)
         except Exception as e:
-            print e
+            print(e)
 
         self.rebuildLock = False
 
     # Reload the menufiles from the filesystem
     def loadMenuFiles( self ):
-        self.menuFiles = []
-        for mainitems in [ "mate-applications.menu", "mate-settings.menu" ]:
-            self.menuFiles.append( Menu( mainitems) )
+        if len(self.menuFiles) == 0:
+            for mainitems in [ "mate-applications.menu", "mate-settings.menu" ]:
+                self.menuFiles.append( Menu( mainitems) )
+
+        for menu in self.menuFiles:
+		    menu.tree.load_sync()
```

Comment:
Not sure if `load_sync` should be called here. We have `Menu` objects recreated a few lines above, and `load_sync` is already called in `Menu`'s constructor...